evolver/device.py

```diff
@@ -71,19 +72,27 @@ def schema(self):
             "controllers": [{"kind": str(type(a)), "config": a.Config.model_json_schema()} for a in self.controllers],
         }
 
+    def _loop_exception_wrapper(self, callable, message="uknown"):
+        try:
+            callable()
+        except Exception as e:
+            if self.raise_loop_exceptions:
+                raise e
+            self.logger.error(f"Error in loop: {message}: {e}")
```

Comment:
Would it not be better to always log here?
```suggestion
            self.logger.error(f"Error in loop: {message}: {e}")
            if self.raise_loop_exceptions:
                raise e
```

Reply:
Also, ``logger.exception`` would be better as it includes the traceback.